src/main/java/com/google/sps/servlets/GenerateServlet.java

```diff
@@ -79,54 +93,89 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     return filteredDestinations;
   }
 
-  /* TODO: Create hunt item / scavenger hunt objects and store in DataStore. */
-  public void writeToDataStore(Set<Destination> filteredDestinations) {
-    System.out.println(filteredDestinations);
+  /* Convert all Destinations in set to Hunt Items. */
+  public ArrayList<HuntItem> convertToHuntItems(Set<Destination> allDestinations) {
+    ArrayList<HuntItem> filteredHuntItems = new ArrayList();
+    for (Destination destination : allDestinations) {
+      HuntItem item = destination.convertToHuntItem();
+      filteredHuntItems.add(item);
+    }
+    return filteredHuntItems;
```

Comment:
You don't need to change it if you prefer the for loop but you can eliminate this method and do this logic in one line with my all time fav, streams!

`allDestinations.stream().map(Destination::convertToHuntItem).collect(Collectors.toList())`

The `map(Destination::convertToHuntItem)` part is equivalent to `map(destination -> destination.convertToHuntItem())`